### ISSUE TITLE
Using with TypeORM

### ISSUE THREAD

**Issue by eylamf** [ospfranco/react-native-quick-sqlite#78]
#### Description
Hey! Thanks for making this great package. I'm attempting to use it with TypeORM but getting the following error when initializing

<img width="224" alt="Screen Shot 2022-08-13 at 12 21 58 PM" src="https://user-images.githubusercontent.com/25336031/184507902-e1269255-0a82-4a27-abe1-f062703ddca3.png">

1. I have this package, `typeorm`, `reflect-metadata` and the required babel plugins installed (**tried clearing all cache already**)
2. I did the patch mentioned in the readme by adding `"./package.json": "./package.json"` to the TypeORM package.json file

DataSource setup:

```ts
datasource = new DataSource({
    type: 'react-native',
    database: '<db_name>',
    location: 'default',
    // seems like the error is coming from here
    driver: require('react-native-quick-sqlite'),
    entities: [...],
    synchronize: true,
  });
```

Also looks like the latest commit commented out the TypeORM code from the `example` project, so I was wondering if the readme is up to date on the necessary steps for getting setup

#### Additonal info
"react-native": "0.68.2",

**Comment by ospfranco** [comment]
You need to restart metro (and the cache) but cannot tell you more, I got it working once with the patch exporting the `package.json` on the TypeORM package.json, but you will have to debug any further issues yourself.

**Comment by ospfranco** [comment]
Did you solve it?

**Comment by LarenDorr** [comment]
Same error.
But  when I debugging, it throw diffrent error.

Sometimes is:

Error: Requiring module "node_modules/typeorm/browser/driver/react-native/ReactNativeDriver.js", which threw an exception: Error: Requiring unknown module "undefined". If you are sure the module exists, try restarting Metro. You may also want to run `yarn` or `npm install`.

or

Error: Requiring unknown module "undefined". If you are sure the module exists, try restarting Metro. You may also want to run `yarn` or `npm install`.

Version info:
"react-native-quick-sqlite": "^4.0.5"
"typeorm": "^0.3.7"

**Comment by LarenDorr** [comment]
I found reason, because used this babel plugin.
link: https://github.com/leonardfactory/babel-plugin-transform-typescript-metadata
But after delete this plugin, typeorm will throw another error:

[ColumnTypeUndefinedError: Column type for TimeLineSource#id is not defined and cannot be guessed. Make sure you have turned on an "emitDecoratorMetadata": true option in tsconfig.json. Also make sure you have imported "reflect-metadata" on top of the main entry file in your application (before any entity imported).If you are using JavaScript instead of TypeScript you must explicitly provide a column type.]

**Comment by LarenDorr** [comment]
I fount a solution, just add `console.log('AbstractSqliteDriver: ', AbstractSqliteDriver);` to `node_modules/typeorm/browser/driver/react-native/ReactNativeDriver.js`. Let `AbstractSqliteDriver` variable to be used.
But I don't know why?? Maybe it has something wrong with compilation.

**Comment by vicary** [comment]
@ospfranco I am using TypeScript and typeorm/typeorm#9178 does not fix it.

Instead I am having this weird error

```
Error: Requiring unknown module "undefined". If you are sure the module exists, try restarting Metro. You may also want to run `yarn` or `npm install`., js engine: hermes

TypeError: Cannot read property 'ReactNativeDriver' of undefined
TypeError: Cannot read property 'ReactNativeDriver' of undefined
```

<details>
<summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/85772/187841974-0ca21d10-c7d1-4285-a9ac-2e40f0d638b0.png)
</details>

More digging into sources in node_modules I found that the most likely cause is metro trying to hoist the optional `require("react-native-sqlite-storage")` and some recursive requiring gives `undefined` in the way.

My current workaround is to also install `react-native-sqlite-storage` so it is able to move pass the require phase, but resolves to `react-native-quick-sqlite` in runtime.

This is sub-optimal because hoisting almost suggest the bundling of `react-native-sqlite-storage` which is never used.

Is there a better way to do it?

**Comment by ospfranco** [comment]
No idea sorry, no time to debug issues right now

**Comment by ospfranco** [comment]
Yeah, this is a problem with Metro and the hardcoded requires of TypeORM, I already tried to debug it and found the instructions on the README working, but not using the library at the moment.

**Comment by vicary** [comment]
OK I'll stick with my workaround for the time being, thanks for replying!

**Comment by elliotsayes** [comment]
If you don't want to edit your node_modules, I created a patch of TypeORM with a couple of the fixes mentioned: https://github.com/typeorm/typeorm/compare/master...elliotsayes:react-native-typeorm:dev
This also adds hacky support for Buffer (sqlite "blob") to the react-native driver. Add to your deps like so:
```
"dependencies": {
    "typeorm": "https://github.com/elliotsayes/react-native-typeorm/blob/574e0b241fa61a1bb371ae553f497e84cd94e7a2/build/typeorm-0.3.10.tgz?raw=true"
}
```

**Comment by ospfranco** [comment]
I have a bit of free time at the moment, looking into this.

I managed to get metro to correctly link typeorm by adding the "./package.json" export on the typeorm root package.json file.

I'm now facing the `ReactNativeDriver` driver issue, I don't remember how I patched it but I think I can create a babel alias for the package in order to trick typeorm to think sqlite-storage is installed.

<img width="1969" alt="Screenshot 001335" src="https://user-images.githubusercontent.com/1634213/206666097-df89fd0b-084e-4ead-ac64-ee233d9e4aa8.png">
